kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -358,6 +358,23 @@ default OwnerReference addOwnerReference(HasMetadata owner) {
           + getKind());
     }
 
+    if (!(owner instanceof GenericKubernetesResource)
+        && !(this instanceof GenericKubernetesResource)
+        && owner instanceof Namespaced) {
+      if (!(this instanceof Namespaced)) {
+        throw new IllegalArgumentException(
+            "Cannot add owner reference from a cluster scoped to a namespace scoped resource: "
+                + optionalMetadata().map(m -> "'" + m.getName() + "' ").orElse("unnamed ")
+                + getKind());
+      } else if (optionalMetadata().map(m -> !Objects.equals(m.getNamespace(), owner.getMetadata().getNamespace()))
+          .orElse(false)) {
+        throw new IllegalArgumentException(
+            "Cannot add owner reference between to resource in a different namespace:"
```

Comment:
Sorry for nitpick but this sentence doesn't sound clear to me. Did you mean this?
```suggestion
            "Cannot add owner reference between two resources in a different namespaces:"
```

Reply:
absolutely, thx!!